docs/linting/Typed_Linting.md

```diff
@@ -28,14 +27,38 @@ module.exports = {
 
 In more detail:
 
-- `parserOptions.tsconfigRootDir` tells our parser the absolute path of your project's root directory.
-- `parserOptions.project` tells our parser the relative path where your project's `tsconfig.json` is.
+- `parserOptions.project` tells our parser to use the closest `tsconfig.json` to each file for informing that file's type information.
   - If your project is a multi-package monorepo, see [our docs on configuring a monorepo](./typed-linting/Monorepos.md).
 - `plugin:@typescript-eslint/recommended-requiring-type-checking` is another recommended configuration we provide. This one contains rules that specifically require type information.
 
 With that done, run the same lint command you ran before.
 You may see new rules reporting errors based on type information!
 
+## Specifying TSConfigs
+
+The `parserOptions.project` option can be turned on with either:
+
+- `true`: to always use `tsconfig.json`s nearest to source files
+- `string | string[]`: any number of glob paths to match TSConfig files relative to the
+
+For example, if you use a specific `tsconfig.eslint.json` for linting, you'd specify:
+
+```js title=".eslintrc.js"
+module.exports = {
+  // ...
+  parserOptions: {
+    project: './tsconfig.eslint.json',
+  },
+  // ...
+};
+```
+
+See [the `@typescript-eslint/parser` docs for more details](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/README.md#parseroptionsproject).
+
+:::note
+If your project is a multi-package monorepo, see [our docs on configuring a monorepo](./typed-linting/Monorepos.md).
+:::
```

Comment:
Btw @bradzacher I just rediscovered https://github.com/typescript-eslint/typescript-eslint/issues/2094#issuecomment-661977512 - kind of cool to see the same idea pop up again:

> I was thinking that we can add a new variant of `parserOptions.project` - the value `true`.
>
> When `parserOptions.project === true`, we shall attempt to automatically find the `tsconfig.json` for the given file.

> if `parserOptions.project === true` - perform the new codepath
> ...

Do you think this PR conflicts with that? Or can we just shunt the described _new codepath_ behind `EXPERIMENTAL_useSourceOfProjectReferenceRedirect`?